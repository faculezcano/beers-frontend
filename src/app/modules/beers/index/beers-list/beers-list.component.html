<div *ngIf="!loading">
  <app-list-item *ngFor="let beer of beers" [beer]="beer"></app-list-item>
</div>

<div *ngIf="loading" class="text-center">
  <span
    class="spinner-border text-muted"
    role="status"
    aria-hidden="true"
  ></span>
</div>
