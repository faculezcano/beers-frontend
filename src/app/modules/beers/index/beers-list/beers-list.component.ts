import { Component, Input, OnInit } from '@angular/core';
import { BeersService } from '../../../../services/beers.service';

@Component({
  selector: 'app-beers-list',
  templateUrl: './beers-list.component.html',
  styleUrls: ['./beers-list.component.scss'],
})
export class BeersListComponent implements OnInit {
  @Input() clearWhenNoParams: boolean = false;
  @Input() loadWithoutParams: boolean = true;
  @Input() loadOnInit: boolean = true;

  private _params: any = {};
  @Input() set params(params: any) {
    this._params = params;
    if (Object.keys(params).length != 0 || this.loadWithoutParams) {
      this.loadBeers();
    } else {
      if (this.clearWhenNoParams) {
        this.clear();
      }
    }
  }
  get params() {
    return this._params;
  }

  public beers: any[] = [];
  public loading = false;

  constructor(private beersService: BeersService) {}

  ngOnInit(): void {
    if (this.loadOnInit) {
      this.loadBeers();
    }
  }

  loadBeers() {
    this.loading = true;
    this.beersService.getBeers(this.params).subscribe((beers) => {
      this.beers = beers as any[];
      this.loading = false;
    });
  }

  clear() {
    this.beers = [];
  }
}
