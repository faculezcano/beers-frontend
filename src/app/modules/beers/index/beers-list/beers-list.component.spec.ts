import { ComponentFixture, TestBed } from '@angular/core/testing';

import { BeersListComponent } from './beers-list.component';
import { HttpClientTestingModule } from '@angular/common/http/testing';

describe('BeersListComponent', () => {
  let component: BeersListComponent;
  let fixture: ComponentFixture<BeersListComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [BeersListComponent],
      imports: [HttpClientTestingModule],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(BeersListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
