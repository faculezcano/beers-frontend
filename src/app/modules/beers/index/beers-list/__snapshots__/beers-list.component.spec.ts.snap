// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeersListComponent should match snapshot 1`] = `
<app-beers-list
  __ngContext__={[Function LRootView]}
  _params={[Function Object]}
  beers={[Function Array]}
  beersService={[Function BeersService]}
  clearWhenNoParams="false"
  loadOnInit={[Function Boolean]}
  loadWithoutParams={[Function Boolean]}
  loading={[Function Boolean]}
>
  <div
    class="text-center"
  >
    <span
      aria-hidden="true"
      class="spinner-border text-muted"
      role="status"
    />
  </div>
</app-beers-list>
`;
